index.js

```diff
@@ -85,49 +84,24 @@ const nonWindowsSingleCall = async (options = {}) => {
 	const lines = stdout.trim().split('\n');
 	lines.shift();
 
-	let psIndex;
-	let commPosition;
-	let argsPosition;
-
-	const processes = lines.map((line, index) => {
+	return lines.map(line => {
 		const match = psOutputRegex.exec(line);
 		if (match === null) {
 			throw new Error(ERROR_MESSAGE_PARSING_FAILED);
 		}
 
-		const {pid, ppid, uid, cpu, memory} = match.groups;
-
-		const processInfo = {
+		const {pid, ppid, uid, cpu, memory, comm, args} = match.groups;
+		const divided = args.split(' ');
+		return {
 			pid: Number.parseInt(pid, 10),
 			ppid: Number.parseInt(ppid, 10),
 			uid: Number.parseInt(uid, 10),
 			cpu: Number.parseFloat(cpu),
 			memory: Number.parseFloat(memory),
-			name: undefined,
-			cmd: undefined
+			name: comm,
+			cmd: [path.basename(divided[0]), divided.slice(1).join(' ')].join(' ')
```

Comment:
Why is there `path.basename`?

Reply:
Also, should give completely wrong results if string contains `/`.

Reply:
`path.basename` used for isolation real command name. Examples:
**/usr/local/bin/docker** -> **docker**
**/usr/local/bin/docker --arguments -a -b -c** -> **docker  --arguments -a -b -c**

Is this behavior fundamentally wrong?

Reply:
Well, if command got path in it, you will get the basename instead of path,so you won't be able to say repeat the command.

Reply:
Removed **path.basename**
For now stored original command from `ps` output